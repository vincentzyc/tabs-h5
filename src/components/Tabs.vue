<template>
  <div class="wg-tabs clearfix">
    <!-- <ul>
      <li v-for="(tab, key) in item.list" :key="key" @click="locationHref(tab.headLink)">
        <img
          @load="imgload"
          :alt="tab.title"
          :src="tab.headImg"
          width="100%"
          v-show="key === item.activeTab"
          v-if="tab.headImg"
        />
      </li>
    </ul> -->

    <ul class="flex tab-bar-wrap">
      <li class="flex-auto tab-bar-title">
        <span>联通</span>
        <img src="@/assets/img/tablogo1.png" alt="联通" style="width:100%"/>
      </li>
      <li class="flex-auto tab-bar-title">
        <span>电信</span>
        <img src="@/assets/img/tablogo2.png" alt="电信" style="width:100%"/>
      </li>
      <li class="flex-auto tab-bar-title">
        <span>移动</span>
        <img src="@/assets/img/tablogo3.png" alt="移动" style="width:100%"/>
      </li>
    </ul>
    <!-- <template v-for="(tabItem, key) in item.list">
      <div :key="key" v-if="tabItem.loaded" v-show="key === item.activeTab">
        <template v-for="tab in tabItem.list">
          <WidgetItems :item="tab" :key="tab.key" :parentItem="item" />
        </template>
      </div>

    </template> -->
  </div>
</template>

<script setup lang="ts">
// import { ref } from "vue";
// defineProps<{ msg: string }>();

// const tabTitles = ref();
</script>
