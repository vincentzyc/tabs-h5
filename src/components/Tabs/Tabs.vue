<template>
  <div class="wg-tabs clearfix">
    <ul class="flex tab-bar-wrap align-end">
      <li
        class="flex flex-center tab-bar"
        v-for="(item, key) in tabList"
        :class="{ active: activeTab === key }"
        @click="changeTab(key)"
      >
        <span class="tab-bar-title">{{ item.title }}</span>
        <img v-lazy="item.logo" :alt="item.title" class="tab-bar-logo" />
      </li>
    </ul>
    <div v-if="tabList[0].loaded" v-show="activeTab === 0" class="tab-content">
      <Tab1 />
    </div>
    <div v-if="tabList[1].loaded" v-show="activeTab === 1" class="tab-content">
      <Tab2 />
    </div>
    <div v-if="tabList[2].loaded" v-show="activeTab === 2" class="tab-content">
      <Tab3 />
    </div>
  </div>
</template>

<script setup lang="ts">
import { defineAsyncComponent } from "vue";
import tablogo1 from "@/assets/img/tablogo1.png";
import tablogo2 from "@/assets/img/tablogo2.png";
import tablogo3 from "@/assets/img/tablogo3.png";
import { useMainStore } from "@/pinia";

const mainStore = useMainStore();

watch(
  () => mainStore.cjAllData,
  n => {
    if (n) resetCjData(tabList[activeTab]);
  }
);

const Tab2 = defineAsyncComponent(() => import("@/components/Tab2/Tab2.vue"));

const Tab3 = defineAsyncComponent(() => import("@/components/Tab3/Tab3.vue"));

let tabList = $ref([
  {
    title: "联通",
    logo: tablogo1,
    loaded: true,
    tabId: "7sj28x5luu",
  },
  {
    title: "电信",
    logo: tablogo2,
    loaded: false,
    tabId: "ug1e2tb61y",
  },
  {
    title: "移动",
    logo: tablogo3,
    loaded: false,
    tabId: "8nngtvbqu3",
  },
]);

let activeTab = $ref(0);

function resetCjData(tabItem: typeof tabList[0]) {
  const curTab = mainStore.cjAllData?.tabInfo.find(item => tabItem.tabId === item.tabId);
  if (curTab) mainStore.setCjData(curTab);
}

function changeTab(i: number) {
  const tabItem = tabList[i];
  tabItem.loaded = true;
  activeTab = i;
  resetCjData(tabItem);
}
</script>
