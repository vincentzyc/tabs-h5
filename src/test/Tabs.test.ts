import { mount } from '@vue/test-utils'
import { Lazyload } from 'vant';
import { createTestingPinia } from '@pinia/testing'
import Tabs from '@/components/Tabs/Tabs.vue'

test('mount Tabs component', async () => {
  expect(Tabs).toBeTruthy()

  const wrapper = mount(Tabs, {
    global: {
      plugins: [createTestingPinia()],
      directives: {
        lazy: Lazyload as any
      }
    },
  })
  expect(wrapper.html()).toMatchSnapshot()
})
