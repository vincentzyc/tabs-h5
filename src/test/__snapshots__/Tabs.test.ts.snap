// Vitest Snapshot v1

exports[`mount Tabs component 1`] = `
"<div class=\\"wg-tabs clearfix\\">
  <ul class=\\"flex tab-bar-wrap align-end\\">
    <li class=\\"flex flex-center tab-bar active\\"><span class=\\"tab-bar-title\\">联通</span><img alt=\\"联通\\" class=\\"tab-bar-logo\\"></li>
    <li class=\\"flex flex-center tab-bar\\"><span class=\\"tab-bar-title\\">电信</span><img alt=\\"电信\\" class=\\"tab-bar-logo\\"></li>
    <li class=\\"flex flex-center tab-bar\\"><span class=\\"tab-bar-title\\">移动</span><img alt=\\"移动\\" class=\\"tab-bar-logo\\"></li>
  </ul>
  <div class=\\"tab-content\\">
    <!-- <SelectNumber @selected=\\"selectNum\\" showBelongCity showSearchNumber showNumberRule /> -->
    <div class=\\"marquee-single flex\\">
      <ul class=\\"scroll-wrapper\\">
        <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">李**抢得靓号 1305678****</span></li>
        <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">蔡**抢得靓号 1308288****</span></li>
        <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">张**抢得靓号 1326668****</span></li>
        <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">罗**抢得靓号 1665688****</span></li>
        <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none man\\"></span><span class=\\"scroll-text txtover\\">林**抢得靓号 1556688****</span></li>
        <li class=\\"flex align-middle scroll-item\\"><span class=\\"userimg flex-none woman\\"></span><span class=\\"scroll-text txtover\\">周**抢得靓号 1306566****</span></li>
      </ul>
    </div>
    <!-- <img src=\\"@/assets/img/tab1-2.jpg\\" alt=\\"产品说明\\" class=\\"widthfull mg-t10\\" />
  <img src=\\"@/assets/img/tab1-3.jpg\\" alt=\\"产品说明\\" class=\\"widthfull\\" />
  <img alt=\\"产品说明\\" class=\\"widthfull img4\\" v-lazy=\\"img4\\" />
  <div class=\\"text-center expenses-wrap\\">
    <span class=\\"expenses-text\\" @click=\\"showExpenses()\\">资费详情说明</span>
  </div>
  <van-popup v-model:show=\\"showForm\\" class=\\"common-popup\\" style=\\"max-width: 576px\\" teleport=\\"body\\">
    <div class=\\"common-wrap\\" v-if=\\"selectNumItem\\">
      <p class=\\"info-text-number\\"
        >已选择<span class=\\"info-number\\">靓号 {{ selectNumItem.num }}</span> {{ showCity }}</p
      >
      <p class=\\"info-text\\">根据国家手机号卡实名制规定</p>
      <p class=\\"info-text\\">请如实填写信息，以便我们及时为您送达</p>
      <BaseForm :handleNo=\\"selectNumItem.num\\" @submit=\\"submitOrder\\" />
      <Agreement title=\\"填写并提交视为阅读并同意\\" :agrList=\\"agrList\\" v-model:checked=\\"checked\\" />
    </div>
  </van-popup>
  <van-popup class=\\"agreement-popup\\" v-model:show=\\"show\\" round>
    <div class=\\"agreement-wrap\\">{{ expensesText }}</div>
    <div class=\\"agreement-confirm\\" @click=\\"close()\\">我知道了</div>
  </van-popup> -->
  </div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
